Makefile

```diff
@@ -40,6 +46,7 @@ all: check-style test dist
 check-style: webapp/node_modules
 	@echo Checking for style guide compliance
 
+## TODO: Add check-types as well
```

Comment:
What's the result when we run `npm run check-types` right now? Maybe the errors we can take care of while this is open? That way the project is ready to be worked on with other typescript contributions. No worries if you think it doesn't make sense to do that in this PR

Reply:
@mickmister On adding the above script, we were getting various errors. So I think we can do that in another PR.